tests/integrations/langchain/langchain_test.py

```diff
@@ -198,6 +199,12 @@ def test_simple_chain_batch(
     calls = list(client.calls(filter=tsi.CallsFilter(trace_roots_only=True)))
     assert_correct_calls_for_chain_batch(calls)
 
+    log_lines = capsys.readouterr().out
+
+    # one parent call link
+    assert log_lines.count("/shawn/test-project/r/call") == 1
```

Comment:
This style of assertion is very fragile. We should avoid asserting on IO unless we are explicitly trying to test I/O

Reply:
fair, although this kind of test would have caught the issue instead of a user reporting it. But I can remove it if you think that it's too fragile

Reply:
It is just a very roundabout way of testing something. i like the notion of making sure errors are not emitted in stdout. but that should be done globally (checkout `logging_error_check` where i do exactly this)

The "thing" you are trying to fix is trace topology